### DISCUSSION TITLE
Suggestions on PSO Optimization

### DISCUSSION THREAD

**Discussion by mh-tanver** [AstroBridge/BDLensing#153] [Q&A] [answered]
Here is my final plots with reduced chi squared of  1.0153813351116814. I was thinking if my gamma value is too high or okay in this case. Also can someone explain the reconstructed source diagram? 

collab link : https://colab.research.google.com/drive/1g80l66HScR_C-EbyKvK7uafI5sN4GAVV?authuser=1#scrollTo=16fb38e8

[{'theta_E': 1.4969106413957614, 'gamma': 2.8654703771051007, 'e1': -0.05394901864551122, 'e2': 0.02811424776334188, 'center_x': -0.0681819499983127, 'center_y': -0.34181292062149543}, {'gamma1': 0.15153575422495777, 'gamma2': -0.012747551314691737, 'ra_0': 0, 'dec_0': 0}] lens result
[{'amp': 1, 'R_sersic': 0.3612717921219933, 'n_sersic': 1.0, 'e1': 0.0969275176137999, 'e2': 0.1828055255626787, 'center_x': -0.12375664900775017, 'center_y': -1.0264015266015045}, {'amp': 1, 'n_max': 6, 'beta': 0.05088588126344952, 'center_x': -0.12375664900775017, 'center_y': -1.0264015266015045}] source result
[{'amp': 1, 'R_sersic': 0.6906875302773381, 'n_sersic': 4.0, 'e1': -0.05394901864551122, 'e2': 0.02811424776334188, 'center_x': -0.0681819499983127, 'center_y': -0.34181292062149543}] 
![download (4)](https://github.com/ajshajib/BDLensing/assets/93901410/6db4b300-54b4-45b5-b541-d47e4d7a6f12)
![download](https://github.com/ajshajib/BDLensing/assets/93901410/e360e0c8-88ff-44ca-aab7-ab1403d245ba)

**Comment by SulymanHossain** [comment, marked as answer]
- I think, your fitted `gamma` value is larger than expected. It should be around 2 for this type of lensing system. Try to model by fixing `gamma=2` and later you can free it by setting a small interval around 2. For example 1.8 <`gamma`<2.2.

- Another thing is, the `R_sersic` value of the source light is enormous compared to the `beta` value. This shouldn't be the case as both parameter describes the amount of source light flux. You can try to limit both parameters to less than 0.2.

- The reconstructed source diagram shows the actual source image though I don't know clearly about the yellow circumscribe.

N.B.: What I understood about the yellow curve from the internet is, It's called lensing caustics. If you remove the "with_caustics=True", this curve will vanish. The area under this closed curve indicates the locations where the source lights are highly magnified. Maybe It's caused by the sharp edges of the lensing galaxy's mass distribution. Besides that, information about the shape of the lens galaxy's mass distribution, the position of the sources, and magnification can be extracted by analyzing the shape of the caustic curve.

**Reply by SulymanHossain** [reply]
@ajshajib bhai, correct me if I'm wrong.
